docs/components/MarkdownPage.js

```diff
@@ -0,0 +1,98 @@
+// ignoring: since we do in fact want to render each component md block again
+// eslint-disable-next-line eslint-comments/disable-enable-pair
+/* eslint-disable react/no-unstable-nested-components */
+// @flow strict
+import { Text, Box } from 'gestalt';
+import { MDXProvider } from '@mdx-js/react';
+
+import Page from './Page.js';
+import PageHeader from './PageHeader.js';
+import MainSection from './MainSection.js';
+
+import { MAX_WIDTH } from './MainSectionSubsection.js';
+
+type Props = {|
+  children: Node,
+  meta: {|
+    title: string,
+    badge: 'pilot' | 'deprecated',
+    component?: boolean,
+    description: string,
+  |},
+  pageSourceUrl?: string,
+|};
+
+export default function MarkdownPage({ children, meta, pageSourceUrl }: Props): Node {
+  const components = {
+    pre: (props) => <MainSection.Card defaultCode={props.children.props.children} />,
+
+    h2: (props) => (
+      <Box marginTop={12} marginBottom={4}>
+        <MainSection name={props.children} />
+      </Box>
+    ),
+    h3: (props) => (
+      <Box
+        ref={(node) => {
+          // eek, hacky. Getting around the .Markdown > h3 coloring css
+          if (node) {
+            node.classList.add('mdx-header');
```

Comment:
Related to the mdx-hedaer css from above. Injecting the css class manually as an ovverride to box. Although, I might be able to use a regular div.

Reply:
Yeah I think we can remove this and not worry about the gray